Automattic/wp-calypso
# GF Signup: Dropdown Interval:  Add dropdown to plan interval toggle behind feature flag

- Related Automattic/growth-foundations#296
## Proposed Changes

Adds drop down to the plan interval toggle instead of the radio multi select buttons

<img width="849" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3422709/2d47bb18-a172-4618-b8f3-85553d424380">

## Not implemented in this PR

- Sticky behaviour in mobile
- Wiring up the experiment
- Maintain dropdown variant on the `/plans` page
- The clickable area around the options is small and need to be fixed (Might have to dive in to the select component)
- The select does not select an option on enter

## Testing Instructions



* Go to `/start?flags=onboarding%2Finterval-dropdown`
* Make sure the dropdown with term interval appears
* Click on a different interval than yearly and make sure the selected plan stays selected
* Select a monthly plan, checkout and make sure the multi year plan selecter discounts matches the discount shown on the plans page


## Pre-merge Checklist
